### PR TITLE
MAT-7051 Add checks for nulls in measure references

### PR DESCRIPTION
**Pull request by ethankaplan**
## MADiE PR Template

Jira Ticket: [MAT-7051](https://jira.cms.gov/browse/MAT-7051)
(Optional) Related Tickets:
https://github.com/MeasureAuthoringTool/measure-service/pull/645
https://github.com/MeasureAuthoringTool/madie-measure/pull/539

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e. Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/models/measure/Reference.java [hunk 1]
```diff
@@ -5,12 +5,22 @@
...
+import jakarta.validation.constraints.NotEmpty;
+import jakarta.validation.constraints.NotNull;
 
 @Data
 @SuperBuilder(toBuilder = true)
 @NoArgsConstructor
 public class Reference {
 	@Id private String id;
 	private String referenceText;
+	@NotNull(
```
Comment (sb-cecilialiu) [inline src/main/java/gov/cms/madie/models/measure/Reference.java]: Do we need separate @NotNull and and @NotEmpty since there is already @NotBlank?

Comment (ethankaplan) [reply src/main/java/gov/cms/madie/models/measure/Reference.java]: You're right, removed

**Review by sb-cecilialiu — APPROVED** [review]
